src/main/java/com/codahale/metrics/WavefrontHistogram.java

```diff
@@ -0,0 +1,166 @@
+package com.codahale.metrics;
+
+import com.google.common.annotations.VisibleForTesting;
+
+import java.io.OutputStream;
+import java.util.List;
+import java.util.function.Supplier;
+
+/**
+ * WavefrontHistogram implementation for com.codahale.metrics
+ * Caveat: Cannot use the same WavefrontHistogram registry for multiple reporters as the reporter
+ * will change the state of the DeltaCounter every time the value is reported.
+ *
+ * @author Sushant Dewan (sushant@wavefront.com).
+ */
+public class WavefrontHistogram extends Histogram implements Metric {
+
+  private final com.wavefront.sdk.entities.histograms.WavefrontHistogram delegate;
```

Comment:
Nit: import vs full qualified name?

Reply:
It actually collides with current com.codahale.metrics.WavefrontHistogram vs 
com.wavefront.sdk.entities.histograms.WavefrontHistogram

Maybe we should rename
com.wavefront.sdk.entities.histograms.WavefrontHistogram to
"WavefrontHistogramImpl"